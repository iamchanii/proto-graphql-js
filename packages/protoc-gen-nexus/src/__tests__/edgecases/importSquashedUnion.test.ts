import { itGeneratesNexusDSLsToMatchSnapshtos } from "../__helpers__/process.test.helper";

describe("import squashed union", () => {
  itGeneratesNexusDSLsToMatchSnapshtos(
    "testapis.edgecases.import_squashed_union.pkg1",
    ["edgecases/import_squashed_union/pkg1/types_pb_nexus.ts"],
  );
  itGeneratesNexusDSLsToMatchSnapshtos(
    "testapis.edgecases.import_squashed_union.pkg2",
    ["edgecases/import_squashed_union/pkg2/types_pb_nexus.ts"],
  );
});
