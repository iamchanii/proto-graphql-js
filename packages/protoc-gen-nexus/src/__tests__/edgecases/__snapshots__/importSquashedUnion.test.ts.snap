// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import squashed union with native protobuf generates nexus DSLs: edgecases/import_squashed_union/pkg1/types_pb_nexus.ts 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg1/types.proto

/* eslint-disable */

import {
  inputObjectType,
  nonNull,
  nullable,
  objectType,
  unionType,
} from "nexus";
import {
  OneofMessage1 as OneofMessage11,
  SquashedOneof as SquashedOneof2,
} from "./testapis/edgecases/import_squashed_union/pkg1/types_pb";

export type testapis$edgecases$import_squashed_union$pkg1$OneofMessage1 =
  OneofMessage11;
export const OneofMessage1 = objectType({
  name: "OneofMessage1",
  definition: (t) => {
    t.field("body", {
      type: nonNull("String"),
      resolve: (source) => {
        const value = source.getBody();
        return value;
      },
      extensions: { protobufField: { name: "body", typeFullName: "string" } },
    });
  },
  isTypeOf: (data: unknown) => {
    return data instanceof OneofMessage11;
  },
  sourceType: {
    module: __filename,
    export: "testapis$edgecases$import_squashed_union$pkg1$OneofMessage1",
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
      name: "OneofMessage1",
      package: "testapis.edgecases.import_squashed_union.pkg1",
    },
  },
});

export const OneofMessage1Input = Object.assign(
  inputObjectType({
    name: "OneofMessage1Input",
    definition: (t) => {
      t.field("body", {
        type: nonNull("String"),
        extensions: { protobufField: { name: "body", typeFullName: "string" } },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        name: "OneofMessage1",
        package: "testapis.edgecases.import_squashed_union.pkg1",
      },
    },
  }),
  {
    toProto: (
      input: NexusGen["inputTypes"]["OneofMessage1Input"],
    ): OneofMessage11 => {
      const output = new OneofMessage11();
      output.setBody(input.body);
      return output;
    },
    _protoNexus: {
      fields: {
        body: {
          type: nonNull("String"),
          extensions: {
            protobufField: { name: "body", typeFullName: "string" },
          },
        },
      },
    },
  },
);

export const SquashedOneofInput = Object.assign(
  inputObjectType({
    name: "SquashedOneofInput",
    definition: (t) => {
      t.field("msg1", {
        type: nullable("OneofMessage1Input"),
        extensions: {
          protobufField: {
            name: "msg1",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
          },
        },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        name: "SquashedOneof",
        package: "testapis.edgecases.import_squashed_union.pkg1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      },
    },
  }),
  {
    toProto: (
      input: NexusGen["inputTypes"]["SquashedOneofInput"],
    ): SquashedOneof2 => {
      const output = new SquashedOneof2();
      if (input.msg1 != null) {
        output.setMsg1(OneofMessage1Input.toProto(input.msg1));
      }
      return output;
    },
    _protoNexus: {
      fields: {
        msg1: {
          type: nullable("OneofMessage1Input"),
          extensions: {
            protobufField: {
              name: "msg1",
              typeFullName:
                "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
            },
          },
        },
      },
    },
  },
);
export const SquashedOneof = unionType({
  name: "SquashedOneof",
  definition: (t) => {
    t.members(OneofMessage1);
  },
  extensions: {
    protobufOneof: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
      name: "SquashedOneof",
      package: "testapis.edgecases.import_squashed_union.pkg1",
      fields: [{
        name: "msg1",
        type: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      }],
    },
  },
});
"
`;

exports[`import squashed union with native protobuf generates nexus DSLs: edgecases/import_squashed_union/pkg2/types_pb_nexus.ts 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg2/types.proto

/* eslint-disable */

import { inputObjectType, nullable, objectType } from "nexus";
import { SquashedOneofInput } from "../pkg1/types_pb_nexus";
import { SquashedOneof } from "./testapis/edgecases/import_squashed_union/pkg1/types_pb";
import { Message as Message1 } from "./testapis/edgecases/import_squashed_union/pkg2/types_pb";

export type testapis$edgecases$import_squashed_union$pkg2$Message = Message1;
export const Message = objectType({
  name: "Message",
  definition: (t) => {
    t.field("msg", {
      type: nullable("SquashedOneof"),
      resolve: (source) => {
        const value = source.getMsg();
        if (value == null) {
          return null;
        }
        switch (value.getMessageCase()) {
          case SquashedOneof.MessageCase.MSG1: {
            return value.getMsg1()!;
          }

          default: {
            return null;
          }
        }
      },
      extensions: {
        protobufField: {
          name: "msg",
          typeFullName:
            "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        },
      },
    });
  },
  isTypeOf: (data: unknown) => {
    return data instanceof Message1;
  },
  sourceType: {
    module: __filename,
    export: "testapis$edgecases$import_squashed_union$pkg2$Message",
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
      name: "Message",
      package: "testapis.edgecases.import_squashed_union.pkg2",
    },
  },
});

export const MessageInput = Object.assign(
  inputObjectType({
    name: "MessageInput",
    definition: (t) => {
      t.field("msg", {
        type: nullable("SquashedOneofInput"),
        extensions: {
          protobufField: {
            name: "msg",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
          },
        },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
        name: "Message",
        package: "testapis.edgecases.import_squashed_union.pkg2",
      },
    },
  }),
  {
    toProto: (input: NexusGen["inputTypes"]["MessageInput"]): Message1 => {
      const output = new Message1();
      if (input.msg != null) {
        output.setMsg(SquashedOneofInput.toProto(input.msg));
      }
      return output;
    },
    _protoNexus: {
      fields: {
        msg: {
          type: nullable("SquashedOneofInput"),
          extensions: {
            protobufField: {
              name: "msg",
              typeFullName:
                "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
            },
          },
        },
      },
    },
  },
);
"
`;

exports[`import squashed union with protobufjs generates nexus DSLs: edgecases/import_squashed_union/pkg1/types_pb_nexus.ts 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg1/types.proto

/* eslint-disable */

import {
  inputObjectType,
  nonNull,
  nullable,
  objectType,
  unionType,
} from "nexus";
import { testapis } from "./testapis/edgecases/import_squashed_union/pkg1";

export type testapis$edgecases$import_squashed_union$pkg1$OneofMessage1 =
  testapis.edgecases.import_squashed_union.pkg1.OneofMessage1;
export const OneofMessage1 = objectType({
  name: "OneofMessage1",
  definition: (t) => {
    t.field("body", {
      type: nonNull("String"),
      resolve: (source) => {
        const value = source.body;
        if (value == null) {
          throw new Error("Cannot return null for non-nullable field");
        }
        return value;
      },
      extensions: { protobufField: { name: "body", typeFullName: "string" } },
    });
  },
  isTypeOf: (data: unknown) => {
    return data instanceof
      testapis.edgecases.import_squashed_union.pkg1.OneofMessage1;
  },
  sourceType: {
    module: __filename,
    export: "testapis$edgecases$import_squashed_union$pkg1$OneofMessage1",
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
      name: "OneofMessage1",
      package: "testapis.edgecases.import_squashed_union.pkg1",
    },
  },
});

export const OneofMessage1Input = Object.assign(
  inputObjectType({
    name: "OneofMessage1Input",
    definition: (t) => {
      t.field("body", {
        type: nonNull("String"),
        extensions: { protobufField: { name: "body", typeFullName: "string" } },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        name: "OneofMessage1",
        package: "testapis.edgecases.import_squashed_union.pkg1",
      },
    },
  }),
  {
    toProto: (
      input: NexusGen["inputTypes"]["OneofMessage1Input"],
    ): testapis.edgecases.import_squashed_union.pkg1.OneofMessage1 => {
      const output = new testapis.edgecases.import_squashed_union.pkg1
        .OneofMessage1();
      output.body = input.body;
      return output;
    },
    _protoNexus: {
      fields: {
        body: {
          type: nonNull("String"),
          extensions: {
            protobufField: { name: "body", typeFullName: "string" },
          },
        },
      },
    },
  },
);

export const SquashedOneofInput = Object.assign(
  inputObjectType({
    name: "SquashedOneofInput",
    definition: (t) => {
      t.field("msg1", {
        type: nullable("OneofMessage1Input"),
        extensions: {
          protobufField: {
            name: "msg1",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
          },
        },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        name: "SquashedOneof",
        package: "testapis.edgecases.import_squashed_union.pkg1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      },
    },
  }),
  {
    toProto: (
      input: NexusGen["inputTypes"]["SquashedOneofInput"],
    ): testapis.edgecases.import_squashed_union.pkg1.SquashedOneof => {
      const output = new testapis.edgecases.import_squashed_union.pkg1
        .SquashedOneof();
      if (input.msg1 != null) {
        output.msg1 = OneofMessage1Input.toProto(input.msg1);
      }
      return output;
    },
    _protoNexus: {
      fields: {
        msg1: {
          type: nullable("OneofMessage1Input"),
          extensions: {
            protobufField: {
              name: "msg1",
              typeFullName:
                "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
            },
          },
        },
      },
    },
  },
);
export const SquashedOneof = unionType({
  name: "SquashedOneof",
  definition: (t) => {
    t.members(OneofMessage1);
  },
  extensions: {
    protobufOneof: {
      fullName: "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
      name: "SquashedOneof",
      package: "testapis.edgecases.import_squashed_union.pkg1",
      fields: [{
        name: "msg1",
        type: "testapis.edgecases.import_squashed_union.pkg1.OneofMessage1",
        options: { "[graphql.object_type]": { squashUnion: true } },
      }],
    },
  },
});
"
`;

exports[`import squashed union with protobufjs generates nexus DSLs: edgecases/import_squashed_union/pkg2/types_pb_nexus.ts 1`] = `
"// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: testapis/edgecases/import_squashed_union/pkg2/types.proto

/* eslint-disable */

import { inputObjectType, nullable, objectType } from "nexus";
import { SquashedOneofInput } from "../pkg1/types_pb_nexus";
import { testapis } from "./testapis/edgecases/import_squashed_union/pkg2";

export type testapis$edgecases$import_squashed_union$pkg2$Message =
  testapis.edgecases.import_squashed_union.pkg2.Message;
export const Message = objectType({
  name: "Message",
  definition: (t) => {
    t.field("msg", {
      type: nullable("SquashedOneof"),
      resolve: (source) => {
        const value = source.msg;
        if (value == null) {
          return null;
        }
        if (value.msg1 != null) {
          return value.msg1;
        }
        return null;
      },
      extensions: {
        protobufField: {
          name: "msg",
          typeFullName:
            "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
        },
      },
    });
  },
  isTypeOf: (data: unknown) => {
    return data instanceof
      testapis.edgecases.import_squashed_union.pkg2.Message;
  },
  sourceType: {
    module: __filename,
    export: "testapis$edgecases$import_squashed_union$pkg2$Message",
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
      name: "Message",
      package: "testapis.edgecases.import_squashed_union.pkg2",
    },
  },
});

export const MessageInput = Object.assign(
  inputObjectType({
    name: "MessageInput",
    definition: (t) => {
      t.field("msg", {
        type: nullable("SquashedOneofInput"),
        extensions: {
          protobufField: {
            name: "msg",
            typeFullName:
              "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
          },
        },
      });
    },
    extensions: {
      protobufMessage: {
        fullName: "testapis.edgecases.import_squashed_union.pkg2.Message",
        name: "Message",
        package: "testapis.edgecases.import_squashed_union.pkg2",
      },
    },
  }),
  {
    toProto: (
      input: NexusGen["inputTypes"]["MessageInput"],
    ): testapis.edgecases.import_squashed_union.pkg2.Message => {
      const output = new testapis.edgecases.import_squashed_union.pkg2
        .Message();
      if (input.msg != null) {
        output.msg = SquashedOneofInput.toProto(input.msg);
      }
      return output;
    },
    _protoNexus: {
      fields: {
        msg: {
          type: nullable("SquashedOneofInput"),
          extensions: {
            protobufField: {
              name: "msg",
              typeFullName:
                "testapis.edgecases.import_squashed_union.pkg1.SquashedOneof",
            },
          },
        },
      },
    },
  },
);
"
`;
